### PR TITLE
feat: Kedro-Airflow convert all pipelines option

### PR DESCRIPTION
**Pull request by sbrugman**
## Description
Closes https://github.com/kedro-org/kedro-plugins/issues/274

## Development notes
The text output of the command was outdated and verbose: 7 lines per DAG. Some of the text is moved to the Readme, other lines are discarded as they seem not relevant (about paths in the template). If needed, we can always choose to add them to the documentation instead of the CLI output.

## Checklist

- [X] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [X] Updated the documentation to reflect the code changes
- [X] Added a description of this change in the relevant `RELEASE.md` file
- [X] Added tests to cover my changes

### CODE REVIEW

**Comment by sbrugman** [comment]
Let me know if there is anything to do on my side to move this forward.

**Comment by ankatiyar** [comment]
Hi @sbrugman, I wanted to ask what the difference is between just running `kedro airflow create` with no `--pipeline` filter vs doing it with `--all`?

File: kedro-airflow/kedro_airflow/__init__.py [hunk 1]
```diff
@@ -1,3 +1,3 @@
 """Kedro plugin for running a project with Airflow."""
 
-__version__ = "0.6.0"
+__version__ = "0.6.1"
```
Comment (ankatiyar) [inline kedro-airflow/kedro_airflow/__init__.py:3]: Could you revert this change? It'll create a new release and we have a separate release process.

**Comment by sbrugman** [comment]
> Hi @sbrugman, I wanted to ask what the difference is between just running `kedro airflow create` with no `--pipeline` filter vs doing it with `--all`?

The default argument for `--pipeline` is the pipeline `__default__`. With `--all`, the conversion takes place for all registered pipelines, which includes `__default__`, and others such as `data_science` and `data_engineering` in some of the starter templates. The `--all` flags generate a DAG for each. This can be used for instance to run them on a different frequency.

I've experimented with changing the default of `--pipeline` to generate all, since users can still use `--pipeline __default__`. Apart from a breaking change, this also seemed unintuitive, so I decided on this approach.

File: kedro-airflow/kedro_airflow/__init__.py (hunk 1 above)
Comment (sbrugman) [reply kedro-airflow/kedro_airflow/__init__.py:3]: Will do!

**Comment by ankatiyar** [comment]
> The default argument for `--pipeline` is the pipeline `__default__`. With `--all`, the conversion takes place for all registered pipelines, which includes `__default__`, and others such as `data_science` and `data_engineering` in some of the starter templates. The `--all` flags generate a DAG for each. This can be used for instance to run them on a different frequency.
> 
> I've experimented with changing the default of `--pipeline` to generate all, since users can still use `--pipeline __default__`. Apart from a breaking change, this also seemed unintuitive, so I decided on this approach.

Could this then not be achieved by allowing a `--pipeline` to accept lists : `kedro airflow create --pipeline=<list all pipelines>` or updating `--pipeline` filter to accept `--pipeline=all` without changing the defaults. What do you think @merelcht @noklam?

**Comment by sbrugman** [comment]
Interesting points.

Accepting a list would be additionally interesting, however would not solve the problem described in the issue.
I could add it anyway, but as extra feature, not as replacement.

`--pipeline=all` would conflict if the user has a pipeline named `all`

**Comment by astrojuanlu** [comment]
I agree with `--pipeline=all` being problematic. What about `--pipeline=*`? Although this does not (yet) exist on Kedro core, for example `kedro run --pipeline=*`.

**Comment by sbrugman** [comment]
Arguments against `--all` are that it introduces a new option and in itself does not refer to `--pipeline`.
Is `*` a valid pipeline name? In that case it could conflict in theory, even though at a low probability.
Reasons for still going with `--all` is that it does not add additional reserved words, nor does it require users to get familiar with the docs: it will just show up in `kedro airflow -h`.
Now that `--pipeline` will support multiple arguments, it will also require less logic to filter out cases such as `--pipeline=de --pipeline=:all:`.

No strong preference on the selected option from my side, as long as we can move forward: this feature will simplify the conversion logic quite a bit.

**Comment by astrojuanlu** [comment]
IIUC, pipeline names must be valid Python identifiers, because they're derived from the subpackage name. Hence both `*` and `:all:` are illegal pipeline names.

I'm also okay with either `--pipeline=:all:` (I suggested this because there was the pip precedent, but I always found it a bit obscure) or adding a separate CLI option `--all`, no strong preference. @ankatiyar @merelcht  what do you think?

**Comment by amandakys** [comment]
Just weighing in, the options that I can see presented here are: 

1. `kedro airflow create --all`
2. `kedro airflow create --pipeline=:all:`
3. `kedro airflow create --pipeline=*`

These commands would create multiple DAG files, one for each registered pipeline. 

My first instinct is that `--all` is the most intuitive solution, if we see the `create` command as describing the functionality of creating DAG files and the `--pipeline` command as taking <pipeline-name> inputs to specify what to target. I don't feel that the lack of `pipeline` in the command would be confusing as the create commands description would be "create DAG files for given pipelines" 

In this vein, I think allowing `--pipeline` to take list input would complement `--all` well. As per our work on plural commands (https://github.com/kedro-org/kedro/issues/2244), I would recommend renaming the command `--pipelines` to indicate the possibility of multiple pipeline inputs. 

On the other hand, we currently have in progress the command `kedro new --add-ons=all` which is similar in concept, but since we control the legal inputs, `all` without modification is okay. With this in mind, Option 3 seems the most consistent, and could become a convention that we extend across our CLI whenever we allow multiple/list inputs to a command and want to allow a shortcut to inputting all values. 

IMO if we aren't interested in rolling out this * syntax and/or see a use case for this (i couldn't find any on a quick skim(, then for this specific issue I would favour the `--all` solution, as it seems the tidiest. 

Some questions I had: 
re: @sbrugman's comment about `--all` being better for discoverability in the CLI as it will come up in `kedro airflow -h` in the case of option 2/3, would it be possible to indicate these special inputs to `--pipeline` in the help interface?

**Comment by sbrugman** [comment]
Extended the CLI option to take multiple `--pipeline` arguments.

For completeness, the help output now is:
```
Usage: Kedro-Airflow airflow create [OPTIONS]

  Create an Airflow DAG for a project

Options:
  -p, --pipeline, --pipelines TEXT
                                  Name of the registered pipeline to convert. If
                                  not set, the '__default__' pipeline is used.
                                  This argument supports passing multiple values
                                  using `--pipeline [p1] --pipeline [p2]`. Use
                                  the `--all` flag to convert all registered
                                  pipelines at once.
  --all                           Convert all registered pipelines at once.
  -e, --env TEXT                  Kedro configuration environment name. Defaults
                                  to `local`.
  -t, --target-dir DIRECTORY      The directory path to store the generated
                                  Airflow dags
  -j, --jinja-file FILE           The template file for the generated Airflow
                                  dags
  --params TEXT                   Specify extra parameters that you want to pass
                                  to the context initialiser. Items must be
                                  separated by comma, keys - by colon or equals
                                  sign, example: param1=value1,param2=value2.
                                  Each parameter is split by the first comma, so
                                  parameter values are allowed to contain
                                  colons, parameter keys are not. To pass a
                                  nested dictionary as parameter, separate keys
                                  by '.', example: param_group.param1:value1.
  --help                          Show this message and exit.
```

**Comment by sbrugman** [comment]
Resolved the merge conflict (pylint)

**Review by astrojuanlu — APPROVED** [review]
Love the new docstring, and happy with the `--all` flag 💯

File: kedro-airflow/RELEASE.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
 # Upcoming Release
 * Added support for Python 3.11
 
+# Release 0.6.1
```
Comment (ankatiyar) [inline kedro-airflow/RELEASE.md:4]: This heading can be removed and the new additions should be under the # Upcoming Release heading. We change these headings from "Upcoming Release" to "Release 0.6.x" when we do a release PR. Could you update this please? 😄

Comment (sbrugman) [reply kedro-airflow/RELEASE.md]: Done!

**Review by ankatiyar — APPROVED** [review]
Thank you @sbrugman for this feature! 💯
